docker/buildx.sh

```diff
@@ -88,14 +93,21 @@ fi
 
 docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
 grep -E -q 'enabled' /proc/sys/fs/binfmt_misc/qemu-arm
-docker run --rm --platform linux/arm64/v8 arm64v8/alpine uname -a |grep -Eq 'aarch64 Linux'
-docker run --rm --platform linux/arm/v7 arm32v7/alpine uname -a|grep -Eq 'armv7l Linux'
+docker run --rm --platform linux/arm64/v8 arm64v8/alpine uname -a | grep -Eq 'aarch64 Linux'
+docker run --rm --platform linux/arm/v7 arm32v7/alpine uname -a | grep -Eq 'armv7l Linux'
 docker buildx create --use --name=ziti-builder 2>/dev/null || docker buildx use --default ziti-builder
 
-eval docker buildx build "${DIRNAME}" \
-    --platform="linux/amd64,linux/arm/v7,linux/arm64" \
-    --build-arg=ZITI_VERSION="${ZITI_VERSION}" \
-    "${TAG_PARAMS}" \
-    --push
-
+# if 
+if [[ ${FLAGS:-} =~ f ]] || ! curl -sSLf https://registry.hub.docker.com/v2/repositories/netfoundry/ziti-edge-tunnel/tags/${ZITI_VERSION} &>/dev/null; then
+    eval docker buildx build "${DIRNAME}" \
```

Comment:
why is this being `eval`ed?

Reply:
@scareything the `eval` is necessary in this case because we're using env var `TAG_PARAMS` to pass command line options to `docker buildx build`. If not eval'd, then quoted, space-separated list of params would be a literal, quoted string as a positional param instead of POSIX arguments and options. It is a choice between not quoting the list (dangerous in a shell script) or using `eval`.

Reply:
Alternatively, you could remove the quotation marks that surround the ${TAG_PARAMS} reference.